src/dissectors/ec_o5logon.c

```diff
@@ -1,43 +1,62 @@
 /*
-    ettercap -- dissector for Oracle O5LOGON protocol -- TCP 1521
-
-    Copyright (C) Dhiru Kholia (dhiru at openwall.com)
-
-    Tested with Oracle 11gR1 and 11gR2 64-bit server and Linux +
-    Windows SQL*Plus clients.
-
-    It does work with Nmap generated packets though the code is hacky.
-
-    This program is free software; you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
-    the Free Software Foundation; either version 2 of the License, or
-    (at your option) any later version.
-
-    This program is distributed in the hope that it will be useful,
-    but WITHOUT ANY WARRANTY; without even the implied warranty of
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-    GNU General Public License for more details.
-
-    You should have received a copy of the GNU General Public License along
-    with this program; if not, write to the Free Software Foundation, Inc.,
-    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
-
-*/
+ * ettercap -- dissector for Oracle O5LOGON protocol -- TCP 1521
+ *
+ * Copyright (c) 2012-2014 Dhiru Kholia (dhiru at openwall.com)
+ * Copyright (c) 2016 magnum
+ *
+ * Tested with Oracle 11gR1 and 11gR2 64-bit server and Linux +
+ * Windows SQL*Plus clients. Now also works with Oracle 12. That
+ * version avoids the known-plain vulnerability from PKCS#7 padding
+ * so more fields are needed for a full attack.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
 
 #include <ec.h>
 #include <ec_decode.h>
 #include <ec_dissect.h>
 #include <ec_session.h>
 
+//#define O5DEBUG
+
 /* globals */
 
 struct o5logon_status {
-   u_char status;
    u_char user[129];
+   u_char srv_addr[22];
+   u_char srv_sk[97];
+   u_char cli_sk[97];
+   u_char pw[256];
+   u_char salt[21];
+   struct {
+      u_int user : 1;
+      u_int c_ano : 1;
+      u_int c_sk : 1;
+      u_int pw : 1;
+      u_int s_ano : 1;
+      u_int vfr : 1;
+      u_int s_sk : 1;
+      u_int pkcs : 2;
+   } flags;
 };
 
-#define WAIT_RESPONSE   1
-#define WAIT_RESULT   2
+/* PKCS#7 padding used? */
+#define MAYBE 0
+#define YES   1
+#define NO    2
```

Comment:
pre-processor directives and variables should be unique. 
please add somwhat like EC_O5LOGON_ to the pre-processor variables.